### PR TITLE
Migrate all the necessary Indy Client Modules

### PR DESCRIPTION
**Pull request by yma88**
ps. Have deleted all deprecated methods in client modules.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.commonjava</groupId>
+    <artifactId>commonjava</artifactId>
+    <version>17</version>
+  </parent>
+
+  <groupId>org.commonjava.indy</groupId>
+  <artifactId>indy-client-modules</artifactId>
```
Comment (ligangty) [inline pom.xml:31]: We should still follow indy client group+artifactId in indy project and versions. We don't want PNC to change this for future usage.

**Comment by ligangty** [comment]
@yma88 I'd recommend rename this project to "indy-client" instead of "indy-client-modules".

**Review by ligangty — COMMENTED** [review]
I think we can still follow the core+addon pattern project structure, because it can let user to decide which module should be imported to remove unnecessary dependencies.

**Comment by yma88** [comment]
> @yma88 I'd recommend rename this project to "indy-client" instead of "indy-client-modules".

Done.

**Comment by yma88** [comment]
> I think we can still follow the core+addon pattern project structure, because it can let user to decide which module should be imported to remove unnecessary dependencies.

@ligangty I update to keep using indy-clients-parent, module: indy-client-core-java, version 3.2.0-SNAPSHOT.
And for the addons module, since we still use the previous packing way for the Content service, addons will also be included, so here it will just focus on packing the indy-*-client-java for the client addons, have updated.
For non-addon client modules, keep them in indy-client-core-java which are the same as before.

**Comment by ligangty** [comment]
> > I think we can still follow the core+addon pattern project structure, because it can let user to decide which module should be imported to remove unnecessary dependencies.
> 
> @ligangty I update to keep using indy-clients-parent, module: indy-client-core-java, version 3.2.0-SNAPSHOT. And for the addons module, since we still use the previous packing way for the Content service, addons will also be included, so here it will just focus on packing the indy-*-client-java for the client addons, have updated. For non-addon client modules, keep them in indy-client-core-java which are the same as before.

Just one question for this client project is: If PNC want to use this, do they need to change anything of current usage except only a version change?

**Comment by yma88** [comment]
> > > I think we can still follow the core+addon pattern project structure, because it can let user to decide which module should be imported to remove unnecessary dependencies.
> > 
> > 
> > @ligangty I update to keep using indy-clients-parent, module: indy-client-core-java, version 3.2.0-SNAPSHOT. And for the addons module, since we still use the previous packing way for the Content service, addons will also be included, so here it will just focus on packing the indy-*-client-java for the client addons, have updated. For non-addon client modules, keep them in indy-client-core-java which are the same as before.
> 
> Just one question for this client project is: If PNC want to use this, do they need to change anything of current usage except only a version change?

@ligangty No need to change the pom dep and package, they are kept the same as before.

**Comment by yma88** [comment]
@yma88

**Review by ligangty — COMMENTED** [review]
Generally LGTM. I think we can merge this as first commit.
@yma88 BTW, can you help add a Jenkinsfile after this PR to enable CI?

**Comment by yma88** [comment]
@ligangty Sure, I will, thanks.

**Pull request merged**